src/slf4ec.c

```diff
@@ -130,7 +130,9 @@ LogResult setLevels(const uint8_t level)
             uint_fast8_t i;
             for (i = 0; i < nbCategories; i++)
             {
-                categories[i]->currentLogLevel = level;
+                if (categories[i]) {
```

Comment:
Is there any benefit for this change?
There's no guarantee the user passed a zeroed-out array on initialization, especially if out-of-bounds. So it doesn't protect against much imho.